profiler/profiler_test.go

```diff
@@ -568,3 +518,32 @@ func TestTelemetryEnabled(t *testing.T) {
 	check("profile_period", time.Duration(10*time.Millisecond).String())
 	check("cpu_duration", time.Duration(1*time.Millisecond).String())
 }
+
+func TestImmediateProfile(t *testing.T) {
```

Comment:
This is what I've come up with so far for testing this change. Time-based tests aren't my favorite but I _think_ this should be okay & not flaky. I'm open to other ideas!